src/sentry/eventstore/snuba/backend.py

```diff
@@ -62,7 +62,9 @@ def get_events(
         )
 
         if "error" not in result:
-            return [SnubaEvent(evt) for evt in result["data"]]
+            events = [SnubaEvent(evt) for evt in result["data"]]
```

Comment:
Could you elaborate on how you noticed the performance issue? Knowing that the timeout is around 30 seconds, I would be surprised if the culprit was fetching multiple columns instead and I would be less surprised if the issue was the select process, which means the queries are still likely to timeout. Could this be related to the max_thread parameter?

Reply:
I think this is more about increasing the speed of existing queries, than reducing the count of timed out queries (hopefully we don't have too many of those). I've seen a number of transactions that look similar to this this https://sentry.io/organizations/sentry/eventsv2/snuba:7c55cef95fef4fafb2444c27376596fe. The 2nd query is significantly slower than the first since it's getting all columns https://sentry.io/organizations/sentry/eventsv2/snuba:7d55be78a00d43329937c42fcba9d2d5. I've changed the get_events implementation to only ever get the minimal columns from Snuba and the rest from Nodestore.